### PR TITLE
Feat/homepage

### PR DESCRIPTION
**Pull request by shaksham08**
Added homepage

### CODE REVIEW

File: src/pages/Homepage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  const [categories, setCategories] = useState([]);
+
+  const getCategories = async () => {
+    try {
+      const { data } = await axios.get("/api/categories");
+      setCategories(data.categories);
+    } catch (error) {
+      console.log(error);
+    }
+  };
```
Comment (tushartiwari7) [inline src/pages/Homepage/index.jsx:18]: Suggestion: you may make a `utils` folder to have all network related requests like getProducts, getCategories, login etc..

File: src/pages/Homepage/index.less [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  padding: 0 5rem;
+}
+
+.hero__container {
+  background: linear-gradient(
+      0deg,
+      var(--black-overlay--light),
+      var(--black-overlay--light)
+    ),
+    url(../../images/shopping.webp);
```
Comment (tushartiwari7) [inline src/pages/Homepage/index.less:11]: liked that you used webp images. 👍

**Review by tushartiwari7 — APPROVED** [review]
UI is awesome with smooth transitions. 
Also CSS looks very clean and managable.
Just some minor suggestions, do have a look.

File: src/pages/Homepage/index.jsx (hunk 1 above)
Comment (shaksham08) [reply src/pages/Homepage/index.jsx:18]: Sure would update it. Thanks

File: src/pages/Homepage/index.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from "react";
+import CategoryCard from "./components/CategoryCard";
+import HeroSection from "./components/HeroSection";
```
Comment (Bhakti199) [inline src/pages/Homepage/index.jsx:3]: you may export this from a single file `index`.

File: src/pages/Homepage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <main>
+      <section className="offer d--flex justify__content--center mt-1">
+        <img src={Banner} alt="40% off" />
+      </section>
+      <HeroSection />
+      <section className="category mt-4 mb-4">
+        <div className="category__container d--flex flex--wrap justify__content--space-between">
+          {categories.map(
+            ({ _id: categoryId, imageUrl, categoryName }, index) => (
```
Comment (Bhakti199) [inline src/pages/Homepage/index.jsx:33]: good one, you destructured it before using.

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me.
- Well written and structured.
- UI looks awesome.
- Proper hierarchy is maintained.
- Proper naming.
- Everything looks good.
- This PR is good to merge.

**Pull request merged**